diff --git a/CMakeLists.txt b/CMakeLists.txt
index 70bb5263..ba23fa1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,12 +45,7 @@ else()
     find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS Crash IconThemes)
 endif()
 
-if(NOT WIN32 AND NOT APPLE AND NOT ANDROID AND NOT HAIKU)
-    find_package(KF6GlobalAccel ${KF_MIN_VERSION} REQUIRED)
-    set(HAVE_KGLOBALACCEL TRUE)
-else()
-    set(HAVE_KGLOBALACCEL FALSE)
-endif()
+set(HAVE_KGLOBALACCEL FALSE)
 
 set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF6KirigamiAddons")
 
