diff --git a/CMakeLists.txt b/CMakeLists.txt
index 70bb5263..adea165c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,15 +42,10 @@ endif()
 if (ANDROID)
     find_package(Gradle REQUIRED)
 else()
-    find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS Crash IconThemes)
+    find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS IconThemes)
 endif()
 
-if(NOT WIN32 AND NOT APPLE AND NOT ANDROID AND NOT HAIKU)
-    find_package(KF6GlobalAccel ${KF_MIN_VERSION} REQUIRED)
-    set(HAVE_KGLOBALACCEL TRUE)
-else()
-    set(HAVE_KGLOBALACCEL FALSE)
-endif()
+set(HAVE_KGLOBALACCEL FALSE)
 
 set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF6KirigamiAddons")
 
